index.bs

```diff
@@ -692,14 +692,33 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. For each |authenticator| currently available on this platform, perform the following steps:
 
-    1. Let |credentialList| be a new [=list=].
-
-    1. If <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code> [=list/is not empty=], execute a 
-        platform-specific procedure to determine which, if any, credentials in
-        <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code> are present on this |authenticator| by matching with 
-        <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}.{{ScopedCredentialDescriptor/id}}</code> and 
-        <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}.{{ScopedCredentialDescriptor/type}}</code>, and set |credentialList| to
-        this filtered list.
+    <span class="note">Note: the procedure to determine the [=authenticators=] present on a platform is 
+        platform-specific.</span>
+
+    1. Let |credentialList| be a new [=ordered set=].
+
+    1. If <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
+        <dl class="switch">
+            : [=list/is not empty=]
+            :: [=set/For each=] credential |C| in <code>|scopedOptions|.{{ScopedCredentialRequestOptions/allowList}}</code>:
+                1. If there is any credential |X| present on this |authenticator| that:
+                    - is mapped to |rpId|, and,
+                    - matches with <code>|C|.{{ScopedCredentialDescriptor/id}}</code> and 
+                        <code>|C|.{{ScopedCredentialDescriptor/type}}</code>,
+
+                    then [=set/append=] |X| to |credentialList|.
```

Comment:
> The next step doesn't actually use the credential for calling authenticatorGetAssertion, just the set of credentials that match allowList

well, if `allowList` is empty, then it calls authenticatorGetAssertion with the set of `ScopedCredentialDescriptor` on that |authenticator| that are associated with rpId, yes?  I.e.,[ |credentialList| is passed to authenticatorGetAssertion](https://w3c.github.io/webauthn/#createCredential). [authenticatorGetAssertion in CTAP takes a sequence](https://fidoalliance.org/specs/fido-v2.0-rd-20161004/fido-client-to-authenticator-protocol-v2.0-rd-20161004.html#authenticatorgetassertion) for that arg (updated CTAP spec will have the type corrected :)

> does it really intend to call authenticatorGetAssertion multiple times on the same authenticator, or would it rather call it once per authenticator or maybe authenticator/transport pair, and let the authenticator choose the credential from the credential list?

yeah, good catch and good questions -- as-written it seems broken.  [authenticatorGetAssertion in CTAP takes a sequence](https://fidoalliance.org/specs/fido-v2.0-rd-20161004/fido-client-to-authenticator-protocol-v2.0-rd-20161004.html#authenticatorgetassertion) for that arg -- so maybe what we should do here is "for each unique `ScopedCredentialDescriptor.tranport` value found in |credentialDescirptorList|, call |authnr| with a sequence of `ScopedCredentialDescriptor` who all have the same `transport` value."